e2e-tests/monitoring-pmm3/run

```diff
@@ -0,0 +1,419 @@
+#!/bin/bash
+
+set -o errexit
+
+test_dir=$(realpath $(dirname $0))
+. ${test_dir}/../functions
+
+set_debug
+
+deploy_pmm3_server() {
+	if [ ! -z "$OPENSHIFT" ]; then
+		platform=openshift
+		oc create sa pmm-server
+		oc adm policy add-scc-to-user privileged -z pmm-server
+		if [[ $OPERATOR_NS ]]; then
+			timeout 30 oc delete clusterrolebinding $(kubectl get clusterrolebinding | grep 'pmm-pxc-operator-' | awk '{print $1}') || :
+			oc create clusterrolebinding pmm-pxc-operator-cluster-wide --clusterrole=percona-xtradb-cluster-operator --serviceaccount=$namespace:pmm-server
+			oc patch clusterrole/percona-xtradb-cluster-operator --type json -p='[{"op":"add","path": "/rules/-","value":{"apiGroups":["security.openshift.io"],"resources":["securitycontextconstraints"],"verbs":["use"],"resourceNames":["privileged"]}}]' -n $OPERATOR_NS
+		else
+			oc create rolebinding pmm-pxc-operator-namespace-only --role percona-xtradb-cluster-operator --serviceaccount=$namespace:pmm-server
+			oc patch role/percona-xtradb-cluster-operator --type json -p='[{"op":"add","path": "/rules/-","value":{"apiGroups":["security.openshift.io"],"resources":["securitycontextconstraints"],"verbs":["use"],"resourceNames":["privileged"]}}]'
+		fi
+		local additional_params="--set platform=openshift --set sa=pmm-server --set supresshttp2=false"
+	fi
+
+	helm uninstall -n "${NAMESPACE}" monitoring || :
+	helm repo remove percona || :
+	kubectl delete clusterrole monitoring --ignore-not-found
+	kubectl delete clusterrolebinding monitoring --ignore-not-found
+	helm repo add percona https://percona.github.io/percona-helm-charts/
+	helm repo update
+
+	retry 10 60 helm install monitoring percona/pmm -n "${NAMESPACE}" \
+		--set fullnameOverride=monitoring \
+		--set image.tag=3.1.0 \
+		--set image.repository=perconalab/pmm-server \
+		--set service.type=LoadBalancer \
+		$additional_params \
+		--force
+}
+
+spinup_pxc() {
```

Comment:
We also have this method in functions and it includes secret. 
I see that in monitoring-2 we use the method from functions. Here you added your own and update secret before cluster created. Is it the idea for it? Can we use method from functions and do not add extra method here?

Reply:
The `spinup_pxc ` from `functions` is creating the secret internally. In our case, we want to have the secret created with a valid PMM3 token before the cluster creation so that once you create the cluster, the PMM containers can spinup properly.

Reply:
What would be if we create the secret and then update it with  the token? I mean if we decided to run pmm after cluster creation. Should we test this case too?

Reply:
> I mean if we decided to run pmm after cluster creation. Should we test this case too?

This will be essentially testing almost the same scenario. In all cases the secret should already exist, update it with the right token and then apply the cr.